src/WebJobs.Script/Binding/Extensibility/ScriptBindingContext.cs

```diff
@@ -35,11 +35,14 @@ public ScriptBindingContext(JObject bindingMetadata)
                     break;
             }
 
-            Name = GetMetadataValue<string>("name");
-            Type = GetMetadataValue<string>("type");
-            DataType = GetMetadataValue<string>("datatype");
-            Cardinality = GetMetadataValue<string>("cardinality");
-            IsTrigger = Type.EndsWith("trigger", StringComparison.OrdinalIgnoreCase);
+            SetOtherMembers();
+        }
+
+        public ScriptBindingContext(JObject bindingMetadata, FileAccess fileAccess)
+        {
+            Metadata = bindingMetadata;
+            Access = fileAccess;
```

Comment:
The code above is already supposed to be deriving the Access from the "direction" member. Why isn't this working for you? Give an example of the function.json for the binding you're trying to get work, or point me to the source code please. I'm not sure this new constructor is the right fix here, and may introduce breaking changes for existing users.

Reply:
@mathewc https://github.com/yojagad/Extensions.ApplicationInsights/tree/master/DummyJsFunc is probably a good example.